application/controllers/UploaderController.php

```diff
@@ -124,6 +124,9 @@ public function run($actionID)
             $filename = $_FILES['uploadfile']['name'];
             // Do we filter file name ? It's used on displaying only , but not save like that.
             //$filename = sanitize_filename($_FILES['uploadfile']['name']);// This remove all non alpha numeric characters and replaced by _ . Leave only one dot .
+            /* Fix file name : need to get only the basename url encoded for display . Avoid usage of basename , need valid UTF8 setlocale*/
+            $aFilename = explode(DIRECTORY_SEPARATOR, rtrim( $filename, DIRECTORY_SEPARATOR));
```

Comment:
This will not work. You can fake the file name to anything you want on upload, even have it contain XSS.
Instead maybe backport my fixed sanitize_filename function from LS5 and use it here.

Reply:
You're right : i close this one.

Previous behaviour are ONLY usage of `basename`